### PR TITLE
[SiliconFlow] Operator fix adaptive_avg_pool2d

### LINKED ISSUE #4915: adaptive_avg_pool2d: window end computed with floor instead of ceil, wrong results for non-divisible input/output sizes
## Summary

`adaptive_avg_pool2d` returns wrong values whenever the input size is not an exact multiple of the output size. The kernel computes the end of each pooling window with **floor**, but PyTorch (ATen) uses **ceil**. As a result, most windows are one element too short, and the average is also divided by the wrong window size. Errors are large (up to ~1.0 on standard normal inputs), not a precision issue.

## Reproduction

```python
import torch
import torch.nn.functional as F
import flag_gems

torch.manual_seed(42)
x = torch.randn(2, 3, 10, 10, device=flag_gems.device)

ref = F.adaptive_avg_pool2d(x.double(), (3, 3)).float()
out = flag_gems.adaptive_avg_pool2d(x, (3, 3))

print((out - ref).abs().max())  # 0.6078 — should be ~1e-7
```

Any 4D input with a non-divisible ratio triggers it, e.g. `224x224 -> 13x13` (max error ~0.07 on the same seed). Exactly divisible ratios (`224 -> 7`, `32 -> 8`, ...) are unaffected, which is why the current test suite passes: every config in `tests/test_adaptive_avg_pool2d.py` uses exactly divisible sizes.

## Root cause

PyTorch defines the pooling window for output index `i` as:

```
start = floor(i * in / out)
end   = ceil((i + 1) * in / out)
```

The kernel computes the end with floor instead
([src/flag_gems/ops/adaptive_avg_pool2d.py#L69-L80](https://github.com/flagos-ai/FlagGems/blob/ac4e5a3dbcc2/src/flag_gems/ops/adaptive_avg_pool2d.py#L69-L80)):

```python
# i_end = floor((i + 1) * in_h / out_h)
h_end = ((h_out_offsets[:, None] + 1) * in_h) // out_h
...
w_end = ((w_out_offsets[None, :] + 1) * in_w) // out_w

# For the last element, extend to the end of input
h_end = tl.where(h_out_offsets[:, None] == out_h - 1, in_h, h_end)
```

The last-element fixup only patches the final index. Every other index whose `(i+1)*in` is not divisible by `out` gets a window that is one element too short, and `window_size` (the divisor) shrinks with it.

Concrete example, `in=10, out=3` (rows):

| output row | PyTorch window | kernel window |
|---|---|---|
| 0 | `[0, 4)` — 4 elements | `[0, 3)` — 3 elements |
| 1 | `[3, 7)` — 4 elements | `[3, 6)` — 3 elements |
| 2 | `[6, 10)` — 4 elements | `[6, 10)` — 4 elements (fixup) |

## Fix

Three changes are needed, not just one:

1. Compute the window end with the ATen ceil form:
   ```python
   h_end = 1 + ((h_out_offsets[:, None] + 1) * in_h - 1) // out_h
   w_end = 1 + ((w_out_offsets[None, :] + 1) * in_w - 1) // out_w
   ```
2. Remove (or enlarge by 1) the `kh < max_win_h` / `kw < max_win_w` masks. Under the ceil rule a window can be one element **longer** than `cdiv(in, out)` — e.g. `in=5, out=3` gives the middle window `[1, 4)`, size 3, while `cdiv(5,3) = 2`. With the masks left as-is, they clip that element and the result is still wrong (verified: max error 0.199 on a `5x5 -> 3x3` case). The masks are redundant anyway, because `h_in < h_end` already bounds every window.
3. The last-element fixup (`tl.where(... == out_h - 1, in_h, ...)`) becomes dead code — the ceil form already yields exactly `in_h` for the last index — and can be removed.

I have this fix working locally (all existing tests plus new non-divisible configs pass, max error vs ATen ~1e-7 in fp32) and can open a PR.

## Environment

- FlagGems master (`ac4e5a3d`); the kernel is unchanged since it was added in #2059
- torch 2.6.0+cu124, triton 3.2.0, NVIDIA H100, CUDA 12.4

### PR DESCRIPTION
**Pull request by fpzh2011**
### PR Category
Operator , OP Test

### Type of Change
Bug Fix

### Description
Fixed the adaptive_avg_pool2d op bug based on issue #4915

### Issue

- Resolves #4915 

### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [ ] Change is fully covered by a UT.

### CODE REVIEW

**Review by huangyiqun — APPROVED** [review]
LGTM.
